src/esp/gfx/PbrImageBasedLighting.cpp

```diff
@@ -0,0 +1,292 @@
+  // we do not use build-in function `renderToTexture` in the CubeMap class. So
+  // we will have to populate the mipmaps by ourselves in this class.
+  environmentMap_ = CubeMap(
+      environmentMapSize,
+      {CubeMap::Flag::ColorTexture | CubeMap::Flag::ManuallyBuildMipmap});
+  irradianceMap_ = CubeMap(irradianceMapSize, {CubeMap::Flag::ColorTexture});
+  prefilteredMap_ = CubeMap(
+      prefilteredMapSize,
+      {CubeMap::Flag::ColorTexture | CubeMap::Flag::ManuallyBuildMipmap});
+}
+
+CubeMap& PbrImageBasedLighting::getIrradianceMap() {
+  CORRADE_ASSERT(flags_ & Flag::IndirectDiffuse,
+                 "PbrImageBasedLighting::getIrradianceMap(): the instance is "
+                 "not created with indirect diffuse part enabled.",
+                 *irradianceMap_);
+
+  CORRADE_ASSERT(
+      irradianceMap_ != Cr::Containers::NullOpt,
+      "PbrImageBasedLighting::getIrradianceMap(): the irradiance map is empty. "
+      "Did you forget to load or compute it (from environment map)?",
+      *irradianceMap_);
+
+  return *irradianceMap_;
+}
+
+CubeMap& PbrImageBasedLighting::getPrefilteredMap() {
+  CORRADE_ASSERT(flags_ & Flag::IndirectSpecular,
+                 "PbrImageBasedLighting::getPrefilteredMap(): the instance is "
+                 "not created with indirect specular part enabled.",
+                 *prefilteredMap_);
+
+  CORRADE_ASSERT(prefilteredMap_ != Cr::Containers::NullOpt,
+                 "PbrImageBasedLighting::getPrefilteredMap(): the pre-filtered "
+                 "cube map is empty."
+                 "Did you forget to load or compute it (from environment map)?",
+                 *prefilteredMap_);
+  return *prefilteredMap_;
+}
+
+Mn::GL::Texture2D& PbrImageBasedLighting::getBrdfLookupTable() {
+  CORRADE_ASSERT(flags_ & Flag::IndirectSpecular,
+                 "PbrImageBasedLighting::getBrdfLookupTable(): the instance is "
+                 "not created with indirect specular part enabled.",
+                 *brdfLUT_);
+
+  CORRADE_ASSERT(brdfLUT_ != Cr::Containers::NullOpt,
+                 "PbrImageBasedLighting::getBrdfLookupTable(): the brdf lookup "
+                 "table (a texture) is empty"
+                 "Did you forget to load or compute it (from environment map)?",
+                 *brdfLUT_);
+
+  return *brdfLUT_;
+}
+
+void PbrImageBasedLighting::loadBrdfLookUpTable() {
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  std::string importerName{"AnyImageImporter"};
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate(importerName);
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  // TODO: HDR, No LDR in the future!
+  // temporarily using the brdflut from here:
+  // https://github.com/SaschaWillems/Vulkan-glTF-PBR/blob/master/screenshots/tex_brdflut.png
+  std::string filename = "./data/pbr/brdflut_ldr_512x512.png";
```

Comment:
Hm, wouldn't it make sense to embed the LUT as a resource, the same way shader sources are? This won't work on Emscripten, for example, and generally it will also break if you don't run from the top-level directory.